modules/candidate_parameters/jsx/monitoring.js

```diff
@@ -0,0 +1,449 @@
+          }
+
+          formattedHistory.push(
+            <tr>
+              <td style={{padding: "1px", width: "25%"}}>
+                <i>{lineInfo}</i>
+              </td>
+              {line}
+            </tr>
+            );
+        }
+      }
+
+      var alertMessage = "";
+      var alertClass = "alert text-center hide";
+      if (this.state.updateResult) {
+        if (this.state.updateResult === "success") {
+          alertClass = "alert alert-success text-center";
+          alertMessage = "Update Successful!";
+        } else if (this.state.updateResult === "error") {
+          var errorMessage = this.state.errorMessage;
+          alertClass = "alert alert-danger text-center";
+          alertMessage = errorMessage ? errorMessage : "Failed to update!";
+        }
+      }
+
+      return (
+        <div class="row">
+          <FormElement
+            name ="monitoringStatus"
+            onSubmit ={this.handleSubmit}
+            ref ="form"
+            class ="col-md-12"
+          >
+            <StaticElement
+              label ="PSCID"
+              text ={this.state.Data.pscid}
+            />
+            <StaticElement
+              label ="DCCID"
+              text ={this.state.Data.candID}
+            />
+            {visits}
+            <div className ={alertClass} role="alert" ref="alert-message">
+              {alertMessage}
+            </div>
+            {updateButton}
+            <table className="table-hist">
+              {formattedHistory}
+            </table>
+
+          </FormElement>
+        </div>
+      );
+    },
+    /**
+     * Handles form submission
+     *
+     * @param {event} e - Form submission event
+     */
+    handleSubmit: function(e) {
+      e.preventDefault();
+      var myFormData = this.state.formData;
+      var today = new Date();
+      var dd = today.getDate();
+      var mm = today.getMonth() + 1; // January is 0!
+      var yyyy = today.getFullYear();
+      if (dd < 10) {
+        dd = '0' + dd;
+      }
+      if (mm < 10) {
+        mm = '0' + mm;
+      }
+      today = yyyy + '-' + mm + '-' + dd;
+      for (let visit in this.state.Data.visits) {
+        if (this.state.Data.visits.hasOwnProperty(visit)) {
+          var label = this.state.Data.visits[visit];
+
+          var dateMonitored = visit + "_date";
+          var dateMonitored2 = visit + "_date2";
+
+          var date1 = myFormData[dateMonitored] ?
+            myFormData[dateMonitored] : null;
+          var date2 = myFormData[dateMonitored2] ?
+            myFormData[dateMonitored2] : null;
+
+          if (date1 !== date2) {
+            alert(label + " dates do not match!");
+            return;
+          }
+          if (date1 > today) {
+            alert(label + " date cannot be later than today!");
+            return;
+          }
+        }
+      }
+
+      // Set form data
+      var self = this;
+      var formData = new FormData();
+      for (var key in myFormData) {
```

Comment:
Especially useful to use `let key` to ensure the var is scoped only inside the loop